src/release_metadata.rs

```diff
@@ -225,3 +227,20 @@ where
         serializer.serialize_none()
     }
 }
+
+async fn get_readme(readme_dir: PathBuf) -> color_eyre::Result<Option<String>> {
+    let mut readme_path: Option<OsString> = None;
+    let mut read_dir = tokio::fs::read_dir(readme_dir).await?;
+
+    while let Some(entry) = read_dir.next_entry().await? {
+        if entry.file_name().to_ascii_lowercase() == README_FILENAME_LOWERCASE {
+            readme_path = Some(entry.file_name());
+        }
+    }
+
+    if let Some(path) = readme_path {
+        Ok(Some(tokio::fs::read_to_string(path).await?))
+    } else {
+        Ok(None)
+    }
```

Comment:
```suggestion
    while let Some(entry) = read_dir.next_entry().await? {
        if entry.file_name().to_ascii_lowercase() == README_FILENAME_LOWERCASE {
            return Some(entry.file_name());
        }
    }

    Ok(None)
```

Reply:
So, this is _almost_ there but earily-returns the filepath rather than the file 😄 But I've fixed that in a commit.